subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/versionupgrade/MergePullRequest.java

```diff
@@ -0,0 +1,94 @@
+package org.shipkit.internal.gradle.versionupgrade;
+
+import java.io.IOException;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.json.simple.JsonArray;
+import org.json.simple.JsonObject;
+import org.json.simple.Jsoner;
+import org.shipkit.internal.gradle.util.BranchUtils;
+import org.shipkit.internal.util.GitHubApi;
+import org.shipkit.internal.util.IncubatingWarning;
+
+class MergePullRequest {
+
+    private static final Logger LOG = Logging.getLogger(MergePullRequest.class);
+
+    public void mergePullRequest(MergePullRequestTask task) throws IOException {
+        mergePullRequest(task, new GitHubApi(task.getGitHubApiUrl(), task.getAuthToken()));
+    }
+
+    public void mergePullRequest(MergePullRequestTask task, GitHubApi gitHubApi) throws IOException {
+        if (task.isDryRun()) {
+            LOG.lifecycle(" Skipping pull request merging due to dryRun = true");
+            return;
+        }
+
+        String headBranch = BranchUtils.getHeadBranch(task.getForkRepositoryName(), task.getVersionBranch());
+
+        IncubatingWarning.warn("merge pull requests");
+        LOG.lifecycle("Waiting for status of a pull request in repository '{}' between base = '{}' and head = '{}'.", task.getUpstreamRepositoryName(), task.getVersionUpgrade().getBaseBranch(), headBranch);
+
+        String sha = retrievePullRequestSha(task, gitHubApi, headBranch);
+        boolean isPending = true;
+        String body = "{" +
+            "  \"head\": \"" + headBranch + "\"," +
+            "  \"base\": \"" + task.getVersionUpgrade().getBaseBranch() + "\"" +
+            "}";
+
+        int timeouts = 0;
+
+        try {
+            while (isPending) {
+                if (timeouts > 20) {
+                    throw new RuntimeException("To many retries. Merge aborted");
+                }
+
+                String statusesResponse = gitHubApi.get("/repos/" + task.getUpstreamRepositoryName() + "/statuses/" + sha);
```

Comment:
Let's extract getting status information into its own class. It will make this class cleaner and remove one of its responsibilities.